eip_operator/src/controller/eip.rs

```diff
@@ -42,11 +47,16 @@ impl k8s_controller::Context for Context {
         client: Client,
         eip: &Self::Resource,
     ) -> Result<Option<Action>, Self::Error> {
-        let eip_api = Api::namespaced(client, &eip.namespace().unwrap());
+        let eip_api = Api::namespaced(client.clone(), eip.namespace().unwrap());
+        let pod_api = Api::<Pod>::namespaced(client.clone(), eip.namespace().unwrap());
+        let node_api = Api::<Node>::all(client.clone());
 
+        // Ensure EIP is Created
         let uid = eip.metadata.uid.as_ref().ok_or(Error::MissingEipUid)?;
         let name = eip.metadata.name.as_ref().ok_or(Error::MissingEipName)?;
         let selector = &eip.spec.selector;
+        let mut status = eip.status.clone().unwrap_or_default();
```

Comment:
this pattern we're moving to here where we just update the mutable status variable is a bit dangerous because if we exit early at any point (with the various uses of `?`, etc) we will lose the updates we have made, which is i think wrong because we are doing things like updating the status after making real changes (disassociating eips, etc). i think we want to make sure to save the status field changes back to the kubernetes api whenever we update this.

Reply:
yeah this is a fair point.  We should patch the resource every time we make a durable change. - code has been updated